### PR TITLE
fix: update cors

### LINKED ISSUE #4155: Add proper cors validation on api and allow specific endpoints

### PR DESCRIPTION
**Pull request by jkcs**
## Related issues

Fixes #4155
/claim #4155

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Explanation of the changes

1702533696014_52C2FD31-5F6B-4ff8-87CB-DD12FDE9D8D7
1702534699695_96E653C8-1481-4679-B010-D2F18FD1DECE
1702534717418_481B17D6-E967-4a4e-B981-825CBB810F5B

### CODE REVIEW

File: apps/api/src/utils/express/corsConfig.ts [hunk 1]
```diff
@@ -1,18 +1,7 @@
...
-      origin.includes('hey.xyz') ||
-      origin.includes('heyxyz.vercel.app') ||
-      origin.startsWith('http://localhost')
-    ) {
-      callback(null, true);
-    } else {
-      callback(new Error('Not allowed by CORS'));
-    }
-  }
+  origin: ['hey.xyz', /\.hey\.xyz$/, /-heyxyz\.vercel\.app$/]
```
Comment (bigint) [inline apps/api/src/utils/express/corsConfig.ts:4]: does it allow localhost?

**Review by bigint — CHANGES_REQUESTED** [review]

File: apps/api/src/utils/express/corsConfig.ts (hunk 1 above)
Comment (jkcs) [reply apps/api/src/utils/express/corsConfig.ts:4]: it's not necessary. localhost:4783 is the API's startup port, and there is no cross-origin issue when accessing it from the same domain.

Comment (jkcs) [reply apps/api/src/utils/express/corsConfig.ts:4]: In general, when debugging locally, cross-origin issues can be resolved by using a proxy to make the requests.

**Comment by bigint** [comment]
https://github.com/heyxyz/hey/issues/4155#issuecomment-1855343284

**Pull request closed without merging**